src/renderer/components/table/__tests__/getSorted.test.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { cloneDeep } from "lodash";
+import { getSorted } from "../sorting";
+
+describe("Table tests", () => {
+  describe("getSorted", () => {
+    it.each([undefined, 5, "", true, {}, []])("should not sort since %j is not a function", () => {
+      expect(getSorted([1, 2, 4, 3], undefined, "asc")).toStrictEqual([1, 2, 4, 3]);
+    });
+
+    it("should sort numerically asc and not touch the original list", () => {
+      const i = [1, 2, 4, 3];
+
+      expect(getSorted(i, v => v, "asc")).toStrictEqual([1, 2, 3, 4]);
+      expect(i).toStrictEqual([1, 2, 4, 3]);
+    });
+
+    it("should sort numerically desc and not touch the original list", () => {
+      const i = [1, 2, 4, 3];
+
+      expect(getSorted(i, v => v, "desc")).toStrictEqual([4, 3, 2, 1]);
+      expect(i).toStrictEqual([1, 2, 4, 3]);
+    });
+
+    it("should sort numerically asc (by defaul) and not touch the original list", () => {
+      const i = [1, 2, 4, 3];
+
+      expect(getSorted(i, v => v, "foobar")).toStrictEqual([1, 2, 3, 4]);
+      expect(i).toStrictEqual([1, 2, 4, 3]);
+    });
+
+    describe("multi-part", () => {
+      it("should treat fewer parts as LESS", () => {
```

Comment:
I don't completely get why fewer parts should be considered as less. Also kubernetes object names don't have dots, does this also work with `-` as part separator?

Reply:
So the `getSorted` function is agnostic to anything to do with kubernetes. It also does not do any splitting itself. That is the the `v => v.split(".")` in this test.

As for this test..., looking at the actual test case, the declared expected sorted version doesn't even satisfy the test case name.

I think I should fix these.